src/factory.js

```diff
@@ -193,6 +194,11 @@ module.exports = ({ Component, createElement, PropTypes }) => {
           ))
           const completeStateKeys = (this._stateKeys = keys(stateDescriptors))
 
+          this._resetState = async () => {
```

Comment:
I'm not sure `async` is supported by every platform this lib supports, please avoid it and return a manual promise: `Promise.resolve()`.

Reply:
I fxed it, but async will be transpiled to generator, right ?

Reply:
It depends of Babel's configuration.

Anyway the transpilation is costly, it generates much bigger code and requires the `regenerator-runtime` library (depending of the target).